### PR TITLE
docs: benchmark standardization, auto-generated API reference, and Rh-enamide matrix

### LINKED ISSUE #121: feat(benchmarks): include JAX MM engine in benchmark suite
## Summary

Add JAX MM engine to the benchmark suite alongside OpenMM and Tinker. Currently `scripts/generate_benchmarks.py` only benchmarks OpenMM and Tinker.

## Tasks

- [ ] Install JAX (`pip install "q2mm[jax]"`) and verify tests pass on WSL
- [ ] Add `JaxEngine` to `scripts/generate_benchmarks.py`
- [ ] Run benchmarks for the same test systems (water, CH₃F)
- [ ] Update `docs/benchmarks/small-molecules.md` with JAX results
- [ ] Update `docs/benchmarks/index.md` summary with JAX data

## Context

JAX provides analytical gradients via `jax.grad` and analytical Hessians via `jax.hessian`, so it should show different performance characteristics than OpenMM (numerical Hessian) and Tinker (subprocess-based).

### PR DESCRIPTION
**Pull request by ericchansen**
## Summary

Comprehensive docs and benchmarking overhaul: auto-generated API reference, benchmark data archival, auto-save CLI, and full Rh-enamide backend × optimizer matrix.

### Benchmark auto-save (feat)
- `BenchmarkResult` now carries `optimized_ff` field with the optimized ForceField
- `save_forcefields()` writes all compatible native formats (.fld, .prm, .xml for MM3)
- CLI `--output` defaults to `benchmark_results/` (was `None`), added `--no-save` flag
- Output organized into `results/` and `forcefields/` subdirectories

### Auto-generated API reference (docs)
- Replaced 25+ manually-maintained reference pages with mkdocs-gen-files auto-generation
- New modules automatically get API docs — no manual steps needed
- Added mkdocs-gen-files, mkdocs-literate-nav, mkdocs-section-index dependencies

### Benchmark data archive (docs)
- CH₃F: 12 JSON results + 36 FF files (4 backends × 3 optimizers × 3 formats)
- Rh-enamide: 12 JSON results (4 backends × 3 optimizers, preliminary 2-iteration runs)
- `benchmarks/README.md` documents the directory layout

### Rh-enamide 4×3 matrix (docs)
Ran all backends × optimizers on the 9-molecule Rh system (182 params):
- **JAX-MD + Nelder-Mead**: 52.3% improvement in 56s (best)
- **L-BFGS-B diverges** on all backends (FD gradients unreliable at 182 params)
- **Powell crashes** with LinAlgError on JAX engines
- **OpenMM/Tinker timeout** — too slow per eval for 182 params
- Filed #147 for scaling improvements

### Doc fixes
- Added Backend column to benchmark tables, sorted by time
- Merged Rh-enamide pages (removed empty Psi4 placeholder)
- Moved API Reference after Benchmarks in nav
- Fixed stale `rh-enamide-jaguar.md` link in tutorial
- Fixed `q2mm.linear_algebra` → `q2mm.models.hessian` references
- Added data links (QM inputs, results, force fields) on all benchmark pages

## Closes

Closes #121

## Test results

- 533 tests pass, lint clean, docs build clean

### CODE REVIEW

File: q2mm/diagnostics/benchmark.py [hunk 1]
```diff
@@ -85,8 +94,76 @@ def from_json(cls, path: str | Path) -> BenchmarkResult:
...
+
+        # Map format name → (saver function, extension, extra kwargs)
+        savers: list[tuple[Any, str, dict[str, Any]]] = []
+        if form_value == "mm3":
+            savers.append((save_mm3_fld, ".fld", {}))
+            savers.append((save_tinker_prm, ".prm", {}))
+            savers.append((save_openmm_xml, ".xml", {"molecule": molecule}))
+        if form_value == "harmonic":
+            savers.append((save_amber_frcmod, ".frcmod", {}))
+
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply q2mm/diagnostics/benchmark.py]: Fixed in d33f029. When `functional_form` is `None`, `save_forcefields()` now returns `[]` with a logged warning instead of silently defaulting to MM3. In practice this won't fire during benchmarks (the engines always set the form), but it's the right guard for programmatic use.
